src/aerialmap_display.cpp

```diff
@@ -462,69 +468,43 @@ void AerialMapDisplay::assembleScene()
   tileCache_.purge({ tileId, blocks_ });
 }
 
-void AerialMapDisplay::transformAerialMap()
+void AerialMapDisplay::transformTileToMapFrame()
 {
-  // We will use five frames in this function:
-  //
-  // * The frame from the NavSatFix message. It is rigidly attached to the robot.
-  // * Rviz's fixed frame (referred to as Ff). It is set by the user in Rviz.
-  // * The ENU  world frame "map".
-  // * The frame of the tiles. We assume that the tiles are in a frame where x points eastwards and y southwards. This
-  //   frame is used by OSM and Google Maps, see
-  //   https://en.wikipedia.org/wiki/Web_Mercator_projection and
-  //   https://developers.google.com/maps/documentation/javascript/coordinates
-  // * The AerialMap frame is an ENU frame. This is where all tiles are placed in. The center tile is in the first
-  //   quadrant, to be exact the left bottom corner of the center tile has coordinates (0,0) in the AerialMap frame.
-
-  if (!isEnabled())
+  auto const currentFixedFrame = context_->getFrameManager()->getFixedFrame();
+
+  // orientation from nav sat fix frame to the map frame
+  Ogre::Quaternion o_navsat_map;
+  // translation from nav sat fix frame to the map frame
+  Ogre::Vector3 t_navsat_map;
+  // temporarily set the fixed frame to map
+  // (Unfortunately the API does not allow looking up arbitrary frame transforms, only towards the currently selected
+  // fixed-frame. Alternatively, we could use the TfBuffer, or work with two transforms (ref_fix to fixed-frame and map
+  // to fixed-frame.)
+  context_->getFrameManager()->setFixedFrame(MAP_FRAME);
```

Comment:
yeah, I was afraid of that too. It looks like it will emit an event and notify all subscribers. In this case maybe even twice, since I change 2 times. Effectively there should not be any damage by this notification, since I change forth an back. But its and overhead to all active plugins and who knows else.

Reply:
Nice!